src/build.d

```diff
@@ -1449,6 +1448,12 @@ auto log(T...)(T args)
         writefln(args);
 }
 
+/// Aborts the current build
+void abortBuild()
+{
+    throw new Exception("Build failed!");
```

Comment:
You may want to throw a specific Exception type for each build failure cause and optionally include additional information.

Reply:
Added the possibility to specify a custom message but custom exceptions for every possible build failure seem overkill IMHO. The only intent of this change is to enable destructors and `scope(...)` statements which will be skipped by `exit`.